tests/test_auth.py

```diff
@@ -75,3 +77,54 @@ def test_handle_407_header_basic(self, extract_cookies_to_jar, proxy_auth_call):
         guess_auth.handle_407(r, *kwargs)
 
         proxy_auth_call.assert_called_with(req)
+
+
+class ProxyAuthHeaderMatcher(betamax.BaseMatcher):
```

Comment:
Tests already exist for HTTPProxyDigestAuth in /tests/test_proxy_digest_auth.py.  It might be beneficial to move these to that file